src/include/miopen/inline_vector.hpp

```diff
@@ -56,7 +58,35 @@ class InlineVector
     InlineVector(const InlineVector& inline_vec)     = default;
     InlineVector(InlineVector&& inline_vec) noexcept = default;
 
-    InlineVector(std::initializer_list<T> data) : real_size(data.size())
+    InlineVector(size_type sz) : real_size(sz)
+    {
+        if(real_size > N)
+        {
+            MIOPEN_THROW("Input data size is bigger than InlineVector's capacity");
+        }
+    }
+
+    InlineVector(const size_type& cnt, const T& val) : real_size(cnt)
+    {
+        if(real_size > N)
+        {
+            MIOPEN_THROW("Input data size is bigger than InlineVector's capacity");
+        }
+
+        std::fill(storage.begin(), storage.begin() + cnt, val);
+    }
+
+    InlineVector(size_type&& cnt, T&& val) : real_size(cnt)
+    {
+        if(real_size > N)
+        {
+            MIOPEN_THROW("Input data size is bigger than InlineVector's capacity");
+        }
+
+        std::fill(storage.begin(), storage.begin() + cnt, val);
+    }
```

Comment:
```suggestion
    InlineVector(const size_typ cnt, const T& val = T()) : real_size(cnt)
    {
        if(real_size > N)
        {
            MIOPEN_THROW("Input data size is bigger than InlineVector's capacity");
        }

        std::fill_n(storage.begin(), cnt, val);
    }
```
A bit more detailed explanation:

It seems that
```C++
    InlineVector(size_type sz) : real_size(sz)
    {
        if(real_size > N)
        {
            MIOPEN_THROW("Input data size is bigger than InlineVector's capacity");
        }
    }
```
May not behave as expected, because when you are constructing `std::vector` like `std::vector<int> foo(5)` it must initialize `foo` with `0` by standard, but you case of `InlineVector` it remains uninitialized.

Also I see not points for rvalue-ref `InlineVector(size_type&& cnt, T&& val)`, btw passing `size_type` by ref or rvalue-ref is not a good idea - most of the time it will work, but it may prevent some optimizations.